comp/dogstatsd/server/batch.go

```diff
@@ -60,6 +62,45 @@ type batcher struct {
 	noAggPipelineEnabled bool
 }
 
+type shardKeyGenerator interface {
```

Comment:
We always use the same generator, we don't need multiple types and an interface to do that, this can be done with just one function instead.